### ISSUE TITLE
julia 1.6 beta

### ISSUE THREAD

**Issue by mdpetters** [JuliaCollections/Memoize.jl#72]
Testing some of my code with julia 1.6-beta throws an error with the ```@memoize``` macro. It's an unusual case with an easy workaround. I am reporting it here because it suggests that something changed under the hood, which might break other use cases as well.  

This construct works fine in julia 1.5.3

```julia
using Memoize, Lazy
@memoize foo(x) = @> x exp sin println
foo(2)
```

which evaluates to 0.8938. In julia 1.6

```julia
using Memoize, Lazy
@memoize foo(x) = @> x exp sin println
```

throws: ```@memoize must be applied to a method definition```

The ```@as``` threading macro from Lazy.jl does work as expected, so it can be used as an easy workaround.

```julia
using Memoize, Lazy
@memoize foo(x) = @as x exp(x) sin(x) println(x)
```

P.S. Tested with Lazy v0.15.1 and Memoize v0.4.4

**Comment by cstjean** [comment]
What if you try `@memoize foo(x) = @>(x, exp, sin, println)`?

`@memoize must be applied to a method definition` means that it's almost certainly either a MacroTools bug, or a Julia bug. It'd be interesting to figure out what changed.

**Comment by cstjean** [comment]
Ouch, 

```julia
julia> :(@> 1 2)
:(#= REPL[8]:1 =# @Error showing value of type Expr:
ERROR: MethodError: no method matching is_syntactic_operator(::String)
Closest candidates are:
  is_syntactic_operator(::Symbol) at show.jl:1221
Stacktrace:
  [1] is_valid_identifier
    @ ./show.jl:1399 [inlined]
  [2] show_sym(io::IOContext{Base.TTY}, sym::String; allow_macroname::Bool)
    @ Base ./show.jl:1438
```

That sounds like something a bug that should be filed in julialang.

**Comment by cstjean** [comment]
It's https://github.com/JuliaLang/julia/issues/39156, but somehow I don't think this will solve the whole problem. Let's wait for the next release...

**Comment by mdpetters** [comment]
Fixed in v1.6.0-rc1.
